src/box/lua/config/applier/credentials.lua

```diff
@@ -526,6 +513,37 @@ local function apply(config)
     create_users(credentials.users)
 end
 
+local function wait_rw(config)
+    if box.info.ro then
+        log.verbose('credentials.apply: Tarantool is in Read Only mode ' ..
+                    'credentials will be set up in the background when it ' ..
+                    'is switched to Read Write mode')
+        box.ctl.wait_rw()
+    end
+    set_credentials(config)
+end
+
+local function apply(config)
+    -- Fiber is required here to have the credentials synced in the background
+    -- when Tarantool is in Read Only mode after it is switched to RW.
+
+    -- Note that only one fiber exists at a time.
+    if fiber_wait_rw == nil or fiber_wait_rw:status() == 'dead' then
+        fiber_wait_rw = fiber.new(wait_rw, config)
+        fiber_wait_rw:set_joinable(true)
+    end
+
+    -- If Tarantool is already in Read Write mode, credentials are still
+    -- applied in the fiber to avoid possible concurrency issues, but the
+    -- main thread is blocked by the `join()`.
+    if not box.info.ro then
+        local ok, err = fiber_wait_rw:join()
+        if not ok then
+            error(err)
+        end
+    end
```

Comment:
I guess we can't mark a fiber as joinable and refuse to join it. All there fibers will wait infinitely for a join call.

Reply:
You are right, let's move `srt_joinable()` to just above the `join()`.

Reply:
Done.